pkg/rbd/nodeserver.go

```diff
@@ -565,6 +568,9 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	if volName != imgInfo.ImageName {
 		return nil, status.Errorf(codes.InvalidArgument, "volume name missmatch between request (%s) and stored metadata (%s)", volName, imgInfo.ImageName)
 	}
+	if imgInfo.Encrypted {
```

Comment:
I don't believe we should allow resize for encrypted volumes at all since it's not supported atm. It might be supported later (https://unix.stackexchange.com/questions/320957/extend-a-luks-encrypted-partition-to-fill-disk) but it is not an easy task to implement hence we should prevent any attempts to do that

I thought this place will capture all the cases, are there any more or only ControllerExpandVolume misssed?

Reply:
ControllerExpandVolume is the first call that expands the rbd image. Later, NodeExpand is called to expand an in-use volume. So if we return said errors in ControllerExpand we should be fine. (NodeExpand will never be called if we error out in ControllerExpand)

Reply:
done, also tested manually and you were right - original check was allowing to expand rbd image but not file system getting stuck in the middle... now it fails out earlier in the process